File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        maxHeap = []

        for i in nums:
            if len(maxHeap) == k:
                heapq.heappushpop(maxHeap, i)
            else:
                heapq.heappush(maxHeap, i)
        # print(maxHeap)
        while k < 1:
            heapq.heappop(maxHeap)
        return maxHeap[0]

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution
{
public:
  int findKthLargest(vector<int> &nums, int k)
  {
    priority_queue<int, vector<int>, greater<int>> pq;
    int count = 0;
    for (int i = 0; i < nums.size(); i++)
    {
      if (count < k)
      {
        pq.push(nums[i]);
        count++;
      }
      else
      {
        if (nums[i] > pq.top())
        {
          pq.pop();
          pq.push(nums[i]);
        }
      }
    }
    return pq.top();
  }
};